tracer/src/Datadog.Trace/Iast/Propagation/PropagationModuleImpl.cs

```diff
@@ -92,98 +94,186 @@ public static void AddTaintedSource(string? input, Source source)
         return result;
     }
 
-    public static object? PropagateResultWhenInputTainted(string? result, object? firstInput, object? secondInput = null, object? thirdInput = null, object? fourthInput = null)
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static bool TryPropagateWholeResult(string? result, object? input, out TaintedObjects? taintedObjects)
```

Comment:
NIT: since this method does not propagate the result if it's different than the input, I would change the name of the method to avoid confusions.